aiida_mlip/data/model.py

```diff
@@ -0,0 +1,125 @@
+"""Define Model Data type in AiiDA"""
+
+import hashlib
+from pathlib import Path
+from urllib import request
+from urllib.parse import urlparse
+
+from aiida.orm import SinglefileData
+
+
+def calculate_hash(file):
+    """Function to calculate file hashes"""
+    # Calculate hash
+    sha256 = hashlib.sha256()
+    with open(file, "rb") as f:
+        for data in iter(lambda: f.read(65536), b""):
+            sha256.update(data)
+    file_hash = sha256.hexdigest()
+    return file_hash
+
+
+def check_existing_file(arch_path, file, file_hash, architecture, cache_path):
```

Comment:
I'll have a think myself, but is there a way to reduce the number of levels of nesting in this function? The more levels, the more difficult it becomes to follow generally.

Reply:
closing this as now it's a bit different anyway following jacob's suggestions, but might be possible to make it better